protocol/app/msgs/nested_msgs.go

```diff
@@ -2,12 +2,17 @@ package msgs
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/authz"
 	gov "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
 )
 
 var (
 	// NestedMsgSamples are msgs that have can include other arbitrary messages inside.
 	NestedMsgSamples = map[string]sdk.Msg{
+		// authz
```

Comment:
Haven't dug into how nested messages are verified - will someone be able to execute an otherwise disallowed messages as a nested message under `MsgExec`? Could you reply to @ttl33 's relevant question [here](https://linear.app/dydx/issue/CORE-794/investigate-and-add-xauthz#comment-f99b5032)?

Reply:
so this PR also updates nested messages verification (see this [commit](https://github.com/dydxprotocol/v4-chain/pull/960/commits/e1b924f7481f848f1bf24a76019a5e1a9013d21c#diff-1a216abd7dbddbbc9b8c42908b044ccf7c982ac2daba515ea6ef10135358aecd)). 

see `validateInnerMsg` for [more details ](https://github.com/dydxprotocol/v4-chain/blob/0c4e1d9771d388fd547e8f1fb2c5a1b41b98d79b/protocol/lib/ante/nested_msg.go#L53C1-L74). Essentially we disallow inner messages to be 
1. unsupported (some ICA controller messages)
2. app injected
3. another nested message